src/wallet/transaction.h

```diff
@@ -192,10 +173,8 @@ class CWalletTx
         mapValue_t mapValueCopy = mapValue;
 
         mapValueCopy["fromaccount"] = "";
-        WriteOrderPos(nOrderPos, mapValueCopy);
-        if (nTimeSmart) {
-            mapValueCopy["timesmart"] = strprintf("%u", nTimeSmart);
-        }
+        if (nOrderPos != -1) mapValueCopy["n"] = ToString(nOrderPos);
+        if (nTimeSmart) mapValueCopy["timesmart"] = strprintf("%u", nTimeSmart);
```

Comment:
Good point considering the potential debugging issue on condition-and-statement-one-liners, never saw it from that perspective (probably because I rarely use debuggers)! Put the braces back and also add the nOrderPos part in the same style.